From 2f2d1f3b441e4671dfe84757071d22fdcbcd2d39 Mon Sep 17 00:00:00 2001
From: Hari Chandrakanthan <quic_haric@quicinc.com>
Date: Tue, 31 Oct 2023 11:38:58 +0530
Subject: [PATCH] wifi: ath12k: optimize dp tx limit threshold

The current global dp tx limit threshold is 49152 and pdev dp tx limit
threshold is 32768. This means that there can be up to 49152 dp skbs
for which tx completion has to be received from hardware.
Higher the number of pending tx skbs, higher the memory consumption
during traffic run. Reduce the memory consumption of the driver
module during traffic run by optimizing the dp tx limit threshold.

The reduced optimum value of global dp tx limit threshold and pdev dp
tx limit threshold are 32000 and 24000 for 1G profile. While 512MB profile 
will have 20000 and 18000 as tx limit threshold. 
With these values, the throughput doesn't get affected.

The pdev dp tx limit threshold does not track ATH12K_NUM_POOL_TX_DESC
anymore. There is no direct relation between optimum pdev tx limit
threshold and ATH12K_NUM_POOL_TX_DESC.

Signed-off-by: Hari Chandrakanthan <quic_haric@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/drivers/net/wireless/ath/ath12k/dp.h
+++ b/drivers/net/wireless/ath/ath12k/dp.h
@@ -288,8 +288,13 @@ struct ath12k_pdev_dp {
 #define DP_REO_QREF_NUM		GENMASK(31, 16)
 #define DP_MAX_PEER_ID		2047
 
-#define ATH12K_DP_PDEV_TX_LIMIT		ATH12K_NUM_POOL_TX_DESC
-#define ATH12K_DP_GROUP_TX_LIMIT	49152
+#ifdef CPTCFG_ATH12K_MEM_PROFILE_512M
+#define ATH12K_DP_PDEV_TX_LIMIT		18000
+#define ATH12K_DP_GROUP_TX_LIMIT	20000
+#else
+#define ATH12K_DP_PDEV_TX_LIMIT		24000
+#define ATH12K_DP_GROUP_TX_LIMIT	32000
+#endif
 
 /* Total size of the LUT is based on 2K peers, each having reference
  * for 17tids, note each entry is of type ath12k_reo_queue_ref
